### ISSUE TITLE
Unfortunate mandatory OSGi import for `jdk.internal.misc` package

### ISSUE THREAD

**Issue by bjhargrave** [joel-costigliola/assertj-core#1663]
#### Summary

The recent update to byte buddy has triggered a new OSGi import for the `jdk.internal.misc` package.

```
$ bnd print -mb assertj-core-3.14.0.jar
[MANIFEST assertj-core-3.14.0]
...
Import-Package                           ...,jdk.internal.misc,...

[USEDBY]
...
jdk.internal.misc                        org.assertj.core.internal.bytebuddy.dynamic.loading
...
```

The `jdk.*` packages should be optionally imported for when you are running on Java 8.

I will submit a PR to fix.

**Comment by joel-costigliola** [comment]
Thanks for reporting this @bjhargrave! We know we are light when it comes to OSGi testing.
Is there an easy way to write a regression test that would capture these errors?

**Comment by bjhargrave** [comment]
We found this issue ourselves because we have an OSGi resolve step in the build of our new osgi-test project. See https://github.com/osgi/osgi-test/blob/eb212d9e6613683d36002962b798a36fce2f5946/org.osgi.test.junit4/pom.xml#L100-L131 for how we found the issue: https://travis-ci.com/osgi/osgi-test/jobs/251664968#L372-L378. 

We could add something similar to the assertj-core build. We may also want to consider migrating the assertj-core build from the older `maven-bundle-plugin` to the `bnd-maven-plugin` from the Bnd devs (of which I am one). If you are interested in any of this, I can work with some of the other Bnd devs to put together a PR or PRs to do this. Let me know.

BTW, the osgi-test project includes some (with more to come) custom AssertJ assertions for OSGi API.
